src/utils/tracked-controls.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Enumerate gamepads and apply callback function to each.
+ *
+ * @param {object} callback - callback function that takes gamepad as argument.
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ */
+module.exports.enumerateGamepads = function (callback, idPrefix) {
+  var gamepads = navigator.getGamepads && navigator.getGamepads();
```

Comment:
in order to be able to perform arbitrary functions for each enumerated gamepad, a callback argument is accepted. otherwise each separate thing one might do needs a new utils function... or ends up coding their own enumerator separately.  is there a more specific objection?